proposals/001-ClusterRepresentation.md

```diff
@@ -0,0 +1,305 @@
+for more than one broker to share an endpoint unless the endpoint type is TLS and the SNI matching address is defined.
+
+Each downstream broker references exactly one upstream broker which must exist within the referenced upstream cluster.  It is an error if there is not a
+one-to-one correspondence between downstream and upstream brokers.
+
+TLS key material may be provided either at the downstream cluster level or at the individual broker level.
+
+
+```yaml
+- name: my-public-cluster
+  upstreamClusterRef: my-private-cluster
+  filterChainRef: my-upstream-filter-chain
+  brokers:
+  - name: broker-1
+    endpointRef: my-tls-endpoint
+    sniMatchAddress: broker-1-public.example.com
+    upstreamBrokerRef: broker-1
+    # or brokers themselves provide the key/cert
+    - tls:
+        key:
+        cert:
+  - name: broker-2
+    ...
+  # cluster wide TLS cert
+  tls:
+    key:
+    cert:
+```
+
+
+#### Upstream Clusters
+
+An _upstream cluster_ represents a physical kafka cluster.   The upstream clusters list specifies all the kafka clustsers that are being exposed through Kroxylicious.  
+
+An _upstream cluster_ may reference a `filter chain`.  This provides zero or more *additional* filters that the RPCs will pass through before passing to the brokers
+of the physical cluster.  
+
+An _upstream cluster_ enumerates the upstream brokers that comprise the physical cluster.
+
+#### Upstream Brokers
+
+Each upstream broker specifies the address of the physical broker and whether TLS is to be used.  It will also specify trust settings.
+
+TLS trust material may be provided at the upstream broker level or at the individual broker level.
+
+```
+- name: my-private-cluster
+  filterChainRef: my-downstream-filter-chain
+  tls:
+    trust:
+    ...
+  brokers:
+  - name: broker-1
+    address: broker-1-private:9092
+    ...
+    tls:
+      trust:
+      ...
+  - name: broker-2
+    address: broker-1-private:9092
+    ...
+    tls:
+      trust:
+```
+
+#### Filter Chain
+
+Currently kroxylicious provide a single filter chain.  It will refactored so that we can support many filter chains, each identified by name. The `downstream cluster` and `upstream cluster` may reference a filter chain. 
+
+```yaml
+filterChains:
+- name: my-upstream-filter-chain
+  filters:
+  - type: ApiVersions
+  - type: BrokerAddress
+
+```
+
+
+### Responsibilities
+
+All specified endpoints will be bound.
+
+When a connection is made to an endpoint,  the system must resolve that connection to a downstream broker and hence a downstream cluster.
+
+To do this, it resolves the endpoint and any SNI information against the model.   This should yield exactly one broker belonging to a downstream cluster.
+It is an error otherwise and the connnection must be closed.
+
+If TLS is in use, the SSLContext can be generated from the downstream cluster definition.  This will be passed to Netty to let it complete the TLS handshake.
```

Comment:
This feels very implementation specific compared to the other responsibilities. 

I see it as an important point to